lib/borda_poll.rb

```diff
@@ -0,0 +1,44 @@
+require 'date'
+
+class BordaPoll
+  class InvalidCandidateError < StandardError
+  end
+
+  class DuplicatedVoterError < StandardError
+  end
+
+  class VoteToExpiredPollError < StandardError
+  end
+
+  attr_reader :title, :candidates, :expiresAt, :votes
+  def initialize(title, candidates, expiresAt = Date.today + 1)
+    @title = title
+    @candidates = candidates
+    @expiresAt = expiresAt
+    @votes = []
+  end
+
+  def add_vote(rankedVote)
+    unless rankedVote.orderedCandidates.sort == candidates.sort
+      raise InvalidCandidateError
+    end
+    if votes.map { |v| v.voter }.include?(rankedVote.voter)
```

Comment:
[nit] `arr.map { |elm| elm.foo }` は `arr.map(&:foo)` と書くのが ruby っぽいです。
詳しくは https://qiita.com/k-penguin-sato/items/420d7487b28b5d58cac4 など。

```suggestion
    if votes.map(&:voter).include?(ranked_vote.voter)
```